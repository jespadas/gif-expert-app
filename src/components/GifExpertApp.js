import React, { useState } from 'react';

import { AddCategory } from './AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    // Usando .concat
    /* const handleAdd = () => {
        setCategories(categories => categories.concat("New category " + Math.random(1)));
    } */

    // Usando Spread ...
    /* const handleAdd = () => {
        setCategories(categories => [...categories, "New category "]);
    } */

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory />
            <hr />



            <ul>
                {categories.map((category) => {
                    return <li key={category}>{category}</li>
                })}
            </ul>

        </>
    )
}