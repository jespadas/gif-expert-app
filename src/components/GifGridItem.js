import React from 'react'

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="animate__animated animate__zoomIn">
            <img
                alt={title}
                src={url}
            />
            <p>{title}</p>
        </div>
    )
}
