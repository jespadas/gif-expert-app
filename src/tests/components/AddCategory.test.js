import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en AddCategory.js', () => {

    // Definir una funcion nativa de jest
    const setCategories = jest.fn();
    let wrapper;

    // Reinicializar valores despues de cada prueba.
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('Debe mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe cambiar la caja de texto', () => {

        const value = 'Hola Mundo';

        const input = wrapper.find('input').at(0);

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').at(0).text().trim()).toBe(value);

    })

    test('No debe de enviar la informacion con submit', () => {

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        })

        expect(setCategories).not.toHaveBeenCalled();

    })

    test('Debe enviar la informacion con submit', () => {

        const value = 'Hola Mundo';

        const input = wrapper.find('input').at(0);

        input.simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        })

        expect(setCategories).toHaveBeenCalled();
        // Comprueba que fue llamada un determinado numero de veces
        expect(setCategories).toHaveBeenCalledTimes(1);
        // Comprueba que fue llamada con argumentos
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    })

    test('Debe de llamar setCategories y limpiar la caja de texto', () => {

        const value = 'Hola';

        // 1 -Simular inputChange
        const input = wrapper.find('input').at(0).simulate('change', { target: { value } });

        // 2 -Simular submit
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        // 3 -setCategories se debe llamar
        expect(setCategories).toHaveBeenCalled();

        // 4 -El valor del input debe ser ''
        expect(wrapper.find('input').prop('value')).toBe('');

    })

})
