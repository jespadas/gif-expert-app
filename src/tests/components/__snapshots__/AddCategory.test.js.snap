// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory.js Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <p>
    Submit a new category: 
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
